src/qt/pivx/masternodeswidget.cpp

```diff
@@ -263,18 +262,17 @@ bool MasterNodesWidget::startMN(CMasternodeConfig::CMasternodeEntry mne, std::st
 
 void MasterNodesWidget::onStartAllClicked(int type)
 {
-    WalletModel::UnlockContext ctx(walletModel->requestUnlock());
-    if (!ctx.isValid()) {
-        // Unlock wallet was cancelled
-        inform(tr("Cannot perform Mastenodes start, wallet locked"));
-        return;
-    }
     if (!checkMNsNetwork()) return;
     if (isLoading) {
         inform(tr("Background task is being executed, please wait"));
     } else {
         isLoading = true;
-        if (!execute(type)) {
+        std::unique_ptr<WalletModel::UnlockContext> pctx = MakeUnique<WalletModel::UnlockContext>(walletModel->requestUnlock());
+        if (!pctx->isValid()) {
+            warn(tr("Start ALL masternodes failed"), tr("Wallet unlock cancelled"));
+            return;
+        }
```

Comment:
put the `isLoading = true` below this bracket line or if the unlock fails, it will be true forever, blocking any subsequent method call.

Reply:
done